tools/build.py

```diff
@@ -0,0 +1,150 @@
+    parser.add_argument("--tag", required=True)
+    args = parser.parse_args()
+
+    assert args.target_platform in ["Linux", "Mac"] and args.conda_path
+
+    # clone repo along with submodules
+    cmd = ["git", "clone", "--recurse-submodules", "https://github.com/isl-org/spear", args.clone_dir]
+    print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd)
+    assert cmd_result.returncode == 0
+
+    os.makedirs(args.output_dir, exist_ok=True)
+
+    # change working directory to the cloned directory above
+    cwd = os.getcwd()
+    print(f"[SPEAR | build.py] Changing current working dir from {cwd} to {args.clone_dir}")
+    os.chdir(args.clone_dir)
+
+    # create a new virtual python env called spear-env
+    if sys.platform == "win32":
+        cmd = ["conda", "create", "-n", "spear-env", "python=3.8", "-y"]
+    elif sys.platform in ["linux", "darwin"]:
+        cmd = ["conda create -n spear-env python=3.8 -y"]
+    else:
+        assert False
+    print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd, shell=True)
+    assert cmd_result.returncode == 0
+
+    # install msgpack-rpc-python package in the newly created python env
+    if sys.platform == "win32":
+        cmd = ["conda", "activate", "spear-env&", "pip", "install", "-e", "third_party/msgpack-rpc-python"]
+    elif sys.platform in ["linux", "darwin"]:    
+        cmd = [f". {args.conda_path}; conda activate spear-env; pip install -e third_party/msgpack-rpc-python"]
+    else:
+        assert False
+    print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd, shell=True)
+    assert cmd_result.returncode == 0
+    
+    # install spear-sim and dependent packages in the newly created python env
+    if sys.platform == "win32":
+        cmd = ["conda", "activate", "spear-env&", "pip", "install", "-e", "python"]
+    elif sys.platform in ["linux", "darwin"]:
+        cmd = [f". {args.conda_path}; conda activate spear-env; pip install -e python"]
+    else:
+        assert False
+    print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd, shell=True)
+    assert cmd_result.returncode == 0
+
+    # change current working directory
+    print(f"[SPEAR | build.py] Changing current working dir from {os.getcwd()} to {os.path.join(args.clone_dir, 'tools')}")
+    os.chdir(os.path.join(args.clone_dir, "tools"))
+
+    if sys.platform == "win32":
+        cmd = ["python", "build_third_party_libs.py", "--num_parallel_jobs", f"{args.num_parallel_jobs}"]
+    elif sys.platform in ["linux", "darwin"]:    
+        cmd = [f"python build_third_party_libs.py --num_parallel_jobs {args.num_parallel_jobs}"]
+    else:
+        assert False
+    print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd, shell=True)
+    assert cmd_result.returncode == 0
+
+    if sys.platform == "win32":
```

Comment:
I think it would be better for this script to do less setup. It should just: clone the repo, create symbolic links, generate a config, build, codesign (mac only), zip. For now, it should assume that everything else is already set up.

Reply:
Agreed, updated the script accordingly.

Reply:
I don't think you need to invoke conda before calling subprocess.run.